lib/fastlane/plugin/json/actions/download_json_action.rb

```diff
@@ -9,13 +9,26 @@ class DownloadJsonAction < Action
       def self.run(params)
         json_url = params[:json_url]
         @is_verbose = params[:verbose]
+        username = params[:username]
+        password = params[:password]
+
         uri = URI(json_url)
         print_params(params) if @is_verbose
         puts("Downloading json from #{uri}") if @is_verbose
 
         begin
-          response = Net::HTTP.get_response(uri)
-          JSON.parse(response.body, symbolize_names: true)
+          Net::HTTP.start(uri.host, uri.port,
+                          use_ssl: uri.scheme == 'https',
+                          verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+
+            request = Net::HTTP::Get.new(uri.request_uri)
+            if !username.nil? && !username.empty? && !password.nil? && !password.empty?
```

Comment:
here you can use `!username.to_s.empty? && !password.to_s.empty?` to make to null or empty check in one :)

Reply:
Yeap, i updated the code as you commented

Reply:
> Just to check, have you tested this with and without authentication, to see that everything is still working?

Yeap, i checked on my Nexus URL with basic auth and it works like charm. But i found no free server to support basic auth to write unit test. Gist only supports signing in via OAuth, then i ignore the unit test

Reply:
Amazing, tomorrow I will add the new version with this changes! Thanks again for this collab!